sns_ik_lib/test/sns_ik_acc_test.cpp

```diff
@@ -0,0 +1,327 @@
+
+  KDL::JntArray aMin = aMax;
+  for (int i = 0; i < int(aMin.rows()); i++) { aMin(i) = -aMin(i); }
+  KDL::JntArray ddqTest = sns_ik::rng_util::getRngBoundedJoints(seed, aMin, aMax);
+
+  // get forward kinematics in acc
+  KDL::Twist accFkTmp;
+  forwardKinematicsAcceleration(qTest, dqTest, ddqTest, pSolver, &accFkTmp);
+
+  // initialize the output for the IK solver
+  KDL::JntArray ddqSolve(qTest.rows());
+  double taskScale;
+
+  // initialize the test result:
+  AccTestResult result;
+
+  // run benchmarking:
+  ros::Time startTime = ros::Time::now();
+  result.exitCode = ikSolver(qTest, dqTest, accFkTmp, ddqSolve, taskScale);
+  
+  result.solveTime += ros::Time::now() - startTime;
+  result.taskScale = taskScale;  // primary task scale
+  KDL::Twist scaledAcc = accFkTmp * result.taskScale;
+  checkForwardKinematicAcceleration(qTest, dqTest, ddqSolve, scaledAcc, pSolver, &(result.fkLinErr), &(result.fkAngErr));
+  result.fkValid = result.fkLinErr <= ACC_IK_TEST_FK_LIN_TOL && result.fkAngErr <= ACC_IK_TEST_FK_ANG_TOL;
+  if (result.fkValid && result.exitCode >= 0) {
+    result.nPass = 1; result.nFail = 0;
+  } else {
+    result.nPass = 0; result.nFail = 1;
+  }
+  return result;
+}
+
+
+/*************************************************************************************************/
+
+/*
+ * @param seed: seed to pass to the RNG on each call
+ *              if seed == 0, then seed is ignored
+ * @param fwdKin: forward kinematics solver for acceleration
+ * @param ikSolver: inverse kinematics solver for acceleration (to be tested)
+ * @param qLow: lower bounds on joints (for generating the test problem)
+ * @param qLow: upper bounds on joints (for generating the test problem)
+ * @param vMax: maximum joint speed (for generating the test problem)
+ * @param aMax: maximum joint acc (for generating the test problem)
+ * @param solverName: solver name, used for logging only
+ * @param nTest: number of tests to run
+ */
+void runGeneralAccIkTest(int seed, sns_ik::SNS_IK pSolver, IkSolver& ikSolver, 
+                  const KDL::JntArray& qLow, const KDL::JntArray& qUpp,
+                  const KDL::JntArray& vMax, const KDL::JntArray& aMax, 
+                  std::string solverName, int nTest = ACC_IK_TEST_COUNT)
+{
+  // Set up the data structure for the results:
+  AccTestResult results;  // accumulate results here
+  AccTestResult tmp; // individual test output here
+
+  // run the test many times and accumulate the results
+  bool success = true;
+  for (int iTest = 0; iTest < nTest; iTest++) {
+    seed++;
+    tmp = runAccIkSingleTest(seed, qLow, qUpp, vMax, aMax, ikSolver, pSolver);
+    if (iTest == 0) {
+      results = tmp;
+    } else {
+      results.nPass += tmp.nPass;
+      results.nFail += tmp.nFail;
+      results.solveTime += tmp.solveTime;
+      results.fkLinErr += tmp.fkLinErr;
+      results.fkAngErr += tmp.fkAngErr;
+      ASSERT_LE(tmp.fkLinErr, ACC_IK_TEST_FK_LIN_TOL);
```

Comment:
Should these asserts also happen when iTest == 0? If so, move them out of the else statement.